lib/experimental/fonts/font-library/class-wp-font-collection.php

```diff
@@ -101,87 +77,95 @@ public function __construct( $slug, $args = array() ) {
+			$data = wp_parse_args(
+				$data,
+				array(
+					'description' => '',
+					'categories'  => array(),
+				)
+			);
+
+			return $data;
 		}
 
 		/**
-		 * Loads the font collection data from a json file path or url.
+		 * Loads the font collection data from a JSON file path or url.
 		 *
 		 * @since 6.5.0
 		 *
-		 * @param string $file_or_url File path or url to a json file containing the font collection data.
+		 * @param string $file_or_url File path or url to a JSON file containing the font collection data.
 		 * @return array|WP_Error An array containing the font collection data on success,
 		 *                        else an instance of WP_Error on failure.
 		 */
-		public static function load_from_json( $file_or_url ) {
+		private function load_from_json( $file_or_url ) {
 			$url  = wp_http_validate_url( $file_or_url );
 			$file = file_exists( $file_or_url ) ? wp_normalize_path( realpath( $file_or_url ) ) : false;
 
 			if ( ! $url && ! $file ) {
-				// translators: %s: File path or url to font collection json file.
-				$message = sprintf( __( 'Font collection JSON file "%s" is invalid or does not exist.', 'gutenberg' ), $file_or_url );
+				// translators: %s: File path or url to font collection JSON file.
+				$message = __( 'Font collection JSON file is invalid or does not exist.', 'gutenberg' );
 				_doing_it_wrong( __METHOD__, $message, '6.5.0' );
 				return new WP_Error( 'font_collection_json_missing', $message );
 			}
 
-			return $url ? self::load_from_url( $url ) : self::load_from_file( $file );
+			return $url ? $this->load_from_url( $url ) : $this->load_from_file( $file );
 		}
 
 		/**
-		 * Loads the font collection data from a json file path.
+		 * Loads the font collection data from a JSON file path.
 		 *
 		 * @since 6.5.0
 		 *
-		 * @param string $file File path to a json file containing the font collection data.
+		 * @param string $file File path to a JSON file containing the font collection data.
 		 * @return array|WP_Error An array containing the font collection data on success,
 		 *                        else an instance of WP_Error on failure.
 		 */
-		private static function load_from_file( $file ) {
-			if ( array_key_exists( $file, static::$collection_json_cache ) ) {
-				return static::$collection_json_cache[ $file ];
-			}
-
+		private function load_from_file( $file ) {
 			$data = wp_json_file_decode( $file, array( 'associative' => true ) );
 			if ( empty( $data ) ) {
 				return new WP_Error( 'font_collection_decode_error', __( 'Error decoding the font collection JSON file contents.', 'gutenberg' ) );
 			}
 
-			if ( empty( $data['slug'] ) ) {
-				// translators: %s: Font collection JSON URL.
-				$message = sprintf( __( 'Font collection JSON file "%s" requires a slug.', 'gutenberg' ), $file );
-				_doing_it_wrong( __METHOD__, $message, '6.5.0' );
-				return new WP_Error( 'font_collection_invalid_json', $message );
-			}
-
-			static::$collection_json_cache[ $file ] = $data;
-
```

Comment:
The slug is now a required arg for both php and json collection registration and validated by the constructor.